GoBGP crashed on Raspberry Pi

I set up my rpi as a route reflector using https://github.com/osrg/gobgp/releases/download/v1.27/gobgp_1.27_linux_armv6.tar.gz binary with a peer on my macbook. In the morning after i awaked my macbook i found GoBGP crashed after the peer goes down. Here is the log and the trace:

```
pi@raspberrypi:~/projects/vxlan $ sudo ./gobgpd -t yaml -f rr-config.yaml                                            
{"level":"info","msg":"gobgpd started","time":"2018-01-27T23:06:42Z"}                                         
{"Topic":"Config","level":"info","msg":"Finished reading the config file","time":"2018-01-27T23:06:42Z"}                
{"level":"info","msg":"PeerGroup rr-client is added","time":"2018-01-27T23:06:42Z"}                                     
{"Name":"rr-client","Topic":"Peer","level":"info","msg":"Add a peer group configuration","time":"2018-01-27T23:06:42Z"} 
{"level":"info","msg":"Dynamic Neighbor 192.168.88.0/24 is added to PeerGroup rr-client","time":"2018-01-27T23:06:42Z"} 
{"Key":"192.168.88.175","State":"BGP_FSM_OPENCONFIRM","Topic":"Peer","level":"info","msg":"Peer Up","time":"2018-01-27T23:07:14Z"}                                                                                                              
{"Key":"192.168.88.175","State":"BGP_FSM_ESTABLISHED","Topic":"Peer","level":"warning","msg":"hold timer expired","time":"2018-01-27T23:27:14Z"}                                                                                                
{"Key":"192.168.88.175","Reason":"hold-timer-expired","State":"BGP_FSM_ESTABLISHED","Topic":"Peer","level":"info","msg":"Peer Down","time":"2018-01-27T23:27:14Z"}                                                                              
{"Code":4,"Data":null,"Key":"192.168.88.175","State":"BGP_FSM_ESTABLISHED","Subcode":0,"Topic":"Peer","level":"warning":
"msg":"sent notification","time":"2018-01-27T23:27:14Z"}                                                                
{"Key":"192.168.88.175","State":"BGP_FSM_OPENCONFIRM","Topic":"Peer","level":"info","msg":"Peer Up","time":"2018-01-27T2
3:35:32Z"}                                                                                                              
{"Key":"192.168.88.175","State":"BGP_FSM_ESTABLISHED","Topic":"Peer","level":"warning","msg":"hold timer expired","time"
:"2018-01-27T23:37:02Z"}                                                                                                
{"Key":"192.168.88.175","Reason":"hold-timer-expired","State":"BGP_FSM_ESTABLISHED","Topic":"Peer","level":"info","msg":
"Peer Down","time":"2018-01-27T23:37:02Z"}                                                                              
{"Code":4,"Data":null,"Key":"192.168.88.175","State":"BGP_FSM_ESTABLISHED","Subcode":0,"Topic":"Peer","level":"warning":
"msg":"sent notification","time":"2018-01-27T23:37:02Z"}                                                                
{"Topic":"Peer","level":"warning","msg":"Cant't find the neighbor 192.168.88.175","time":"2018-01-27T23:37:02Z"}        
{"Key":"192.168.88.175","State":"BGP_FSM_OPENCONFIRM","Topic":"Peer","level":"info","msg":"Peer Up","time":"2018-01-27T23:44:41Z"}
{"Key":"192.168.88.175","State":"BGP_FSM_ESTABLISHED","Topic":"Peer","level":"warning","msg":"hold timer expired","time"
:"2018-01-27T23:46:11Z"}                                                                                                
{"Key":"192.168.88.175","Reason":"hold-timer-expired","State":"BGP_FSM_ESTABLISHED","Topic":"Peer","level":"info","msg":"Peer Down","time":"2018-01-27T23:46:11Z"}
{"Code":4,"Data":null,"Key":"192.168.88.175","State":"BGP_FSM_ESTABLISHED","Subcode":0,"Topic":"Peer","level":"warning",
"msg":"sent notification","time":"2018-01-27T23:46:11Z"}
{"Key":"192.168.88.175","State":"BGP_FSM_OPENCONFIRM","Topic":"Peer","level":"info","msg":"Peer Up","time":"2018-01-27T23:53:52Z"}
{"Key":"192.168.88.175","State":"BGP_FSM_ESTABLISHED","Topic":"Peer","level":"warning","msg":"hold timer expired","time":"2018-01-27T23:55:22Z"}
{"Key":"192.168.88.175","Reason":"hold-timer-expired","State":"BGP_FSM_ESTABLISHED","Topic":"Peer","level":"info","msg":"Peer Down","time":"2018-01-27T23:55:22Z"}
{"Code":4,"Data":null,"Key":"192.168.88.175","State":"BGP_FSM_ESTABLISHED","Subcode":0,"Topic":"Peer","level":"warning","msg":"sent notification","time":"2018-01-27T23:55:22Z"}
{"Key":"192.168.88.175","State":"BGP_FSM_OPENCONFIRM","Topic":"Peer","level":"info","msg":"Peer Up","time":"2018-01-28T00:01:41Z"}
{"Key":"192.168.88.175","State":"BGP_FSM_ESTABLISHED","Topic":"Peer","level":"warning","msg":"hold timer expired","time":"2018-01-28T00:59:11Z"}
{"Key":"192.168.88.175","Reason":"hold-timer-expired","State":"BGP_FSM_ESTABLISHED","Topic":"Peer","level":"info","msg":"Peer Down","time":"2018-01-28T00:59:11Z"}
{"Code":4,"Data":null,"Key":"192.168.88.175","State":"BGP_FSM_ESTABLISHED","Subcode":0,"Topic":"Peer","level":"warning","msg":"sent notification","time":"2018-01-28T00:59:11Z"}
{"Topic":"Peer","level":"warning","msg":"Cant't find the neighbor 192.168.88.175","time":"2018-01-28T00:59:11Z"}
{"Key":"192.168.88.175","State":"BGP_FSM_OPENCONFIRM","Topic":"Peer","level":"info","msg":"Peer Up","time":"2018-01-28T01:06:12Z"}
{"Key":"192.168.88.175","State":"BGP_FSM_ESTABLISHED","Topic":"Peer","level":"warning","msg":"hold timer expired","time":"2018-01-28T01:36:42Z"}
{"Key":"192.168.88.175","Reason":"hold-timer-expired","State":"BGP_FSM_ESTABLISHED","Topic":"Peer","level":"info","msg":"Peer Down","time":"2018-01-28T01:36:42Z"}
{"Code":4,"Data":null,"Key":"192.168.88.175","State":"BGP_FSM_ESTABLISHED","Subcode":0,"Topic":"Peer","level":"warning","msg":"sent notification","time":"2018-01-28T01:36:42Z"}
panic: close of closed channel

goroutine 172 [running]:
github.com/osrg/gobgp/vendor/github.com/eapache/channels.(*InfiniteChannel).Close(...)
        /Users/fujita/Devel/go/src/github.com/osrg/gobgp/vendor/github.com/eapache/channels/infinite_channel.go:40
github.com/osrg/gobgp/server.cleanInfiniteChannel(0x128be800)
        /Users/fujita/Devel/go/src/github.com/osrg/gobgp/server/util.go:30 +0x20
github.com/osrg/gobgp/server.(*Peer).stopFSM(0x12a7aa80, 0x1, 0x1)
        /Users/fujita/Devel/go/src/github.com/osrg/gobgp/server/peer.go:671 +0x53c
created by github.com/osrg/gobgp/server.(*BgpServer).handleFSMMessage
        /Users/fujita/Devel/go/src/github.com/osrg/gobgp/server/server.go:854 +0xea0
```

My RR config is

```
global:
  config:
    as: 65000
    router-id: 192.168.88.177
    local-address-list:
      - 192.168.88.177
peer-groups:
  - config:
      peer-group-name: rr-client
      peer-as: 65000
    afi-safis:
      - config:
          afi-safi-name: l2vpn-evpn
    route-reflector:
      config:
        route-reflector-client: true
        route-reflector-cluster-id: 192.168.88.177
dynamic-neighbors:
  - config:
      peer-group: rr-client
      prefix: 192.168.88.0/24
```

--- a/server/server.go
+++ b/server/server.go
@@ -853,6 +853,8 @@ func (server *BgpServer) handleFSMMessage(peer *Peer, e *FsmMsg) {
 				peer.stopPeerRestarting()
 				go peer.stopFSM()
 				delete(server.neighborMap, peer.fsm.pConf.State.NeighborAddress)
+				server.broadcastPeerState(peer, oldState)
+				return
 			}
 		} else if peer.fsm.pConf.GracefulRestart.State.PeerRestarting && nextState == bgp.BGP_FSM_IDLE {
 			if peer.fsm.pConf.GracefulRestart.State.LongLivedEnabled {

